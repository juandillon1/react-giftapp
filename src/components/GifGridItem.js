export const GifGridItem = ( {title, url} ) => {
    return (
        <div className="card animate__animated animate__bounce animate__slow">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    );
};