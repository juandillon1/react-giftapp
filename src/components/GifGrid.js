import { useFetchGifs } from "../hooks/useFetchGifs";
import { PropTypes } from 'prop-types';
import { GifGridItem } from './GifGridItem';
export const GifGrid = ({categories}) => {
    const {data: images, loading} = useFetchGifs(categories);

    return (
        <>
            <h3 className="animate__animated animate__bounce animate__slow"> {categories} </h3>
            {loading && <p className="animate__animated animate__flash">Cargando...</p> }
            <div className="card-grid">
            {
                images.map(img =>
                    <GifGridItem key={img.id} {...img}/>
                )
            }
            </div>
        </>
    )
};

GifGrid.propTypes = {
    categories: PropTypes.string.isRequired,
}