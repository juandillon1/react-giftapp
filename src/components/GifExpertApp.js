import {useState} from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';
const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Dragon Ball Z']);

    // const handleAdd = () => {
    //     setCategories([...categories, 'Naruto']);
    // };
    return (
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories}/>
        <hr />

        <ol>
            {
                categories.map( categoria =>
                    <GifGrid categories={categoria} key={categoria}/>
                )
            }
        </ol>
        </>
    );
};

export default GifExpertApp;