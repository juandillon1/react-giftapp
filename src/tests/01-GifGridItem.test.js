import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem}  from '../components/GifGridItem';
describe('Componente pruebas', () => {
    const title = 'Un titulo';
    const url = 'https://localhost.com/test.jpg';
    const wrapper = shallow(<GifGridItem title = {title} url = {url}/>);
    test('Debe mostrar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    test('Debe tener la imagen con un src y un alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    test('debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        // expect(div.prop('className')).toContain('animate__slow')
        expect(div.prop('className').includes('animate__slow')).toBe(true)
        // expect(div.prop('className').includes('animate__slow')).not.toBe(true)
    })


});

