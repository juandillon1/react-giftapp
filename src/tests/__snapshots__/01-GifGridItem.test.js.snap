// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente pruebas Debe mostrar el componente 1`] = `
<div
  className="card animate__animated animate__bounce animate__slow"
>
  <img
    alt="Un titulo"
    src="https://localhost.com/test.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
